diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d5bed2..47dbebf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,7 +64,7 @@ FetchContent_Declare(
 )
 # For Windows: Prevent overriding the parent project's compiler/linker settings
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-FetchContent_MakeAvailable(googletest)
+#FetchContent_MakeAvailable(googletest)
 enable_testing()
 
 # Enable Hot Reload for MSVC compilers if supported.
@@ -109,4 +109,4 @@ add_subdirectory("OdbDesignLib")
 add_subdirectory("OdbDesignApp")
 add_subdirectory("OdbDesignServer")
 add_subdirectory("Utils")
-add_subdirectory("OdbDesignTests")
+#add_subdirectory("OdbDesignTests")
diff --git a/OdbDesignLib/CMakeLists.txt b/OdbDesignLib/CMakeLists.txt
index 73a1b59..3cdf7c3 100644
--- a/OdbDesignLib/CMakeLists.txt
+++ b/OdbDesignLib/CMakeLists.txt
@@ -56,12 +56,13 @@ file(GLOB PROTO_FILES "${PROTO_DIR}/*.proto")
 message(STATUS "Found proto files: ${PROTO_FILES}")
 
 # Link to Protobuf
-find_package(Protobuf CONFIG REQUIRED)
+find_package(Protobuf REQUIRED)
 
 # Generate protobuf files with export macro
 protobuf_generate(
     TARGET OdbDesign
     PROTOS ${PROTO_FILES}
+    APPEND_PATH
     IMPORT_DIRS ${PROTO_DIR}
     EXPORT_MACRO ODBDESIGN_EXPORT
 )
@@ -78,3 +79,5 @@ target_link_libraries(OdbDesign PUBLIC protobuf::libprotobuf)
 # (see https://github.com/CrowCpp/Crow/issues/661#issuecomment-1702544225)
 list(REMOVE_ITEM _CROW_ICD "_CROW_ICD-NOTFOUND")
 set_target_properties(Crow::Crow PROPERTIES INTERFACE_COMPILE_DEFINITIONS "${_CROW_ICD}")
+target_include_directories(OdbDesign PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/protoc)
+target_include_directories(OdbDesign PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
diff --git a/OdbDesignLib/IProtoBuffable.h b/OdbDesignLib/IProtoBuffable.h
index 5840312..6435035 100644
--- a/OdbDesignLib/IProtoBuffable.h
+++ b/OdbDesignLib/IProtoBuffable.h
@@ -90,7 +90,7 @@ namespace Odb::Lib
 	inline void IProtoBuffable<TPbMessage>::from_json(const std::string& json)
 	{			
 		auto pMessage = std::make_unique<TPbMessage>();
-		auto status = google::protobuf::util::JsonStringToMessage(absl::string_view(json), pMessage.get());
+		auto status = google::protobuf::util::JsonStringToMessage(json, pMessage.get());
 		if (!status.ok()) return;		
 		from_protobuf(*pMessage);		
 	}
diff --git a/OdbDesignServer/CMakeLists.txt b/OdbDesignServer/CMakeLists.txt
index ba3a6d2..ff3a1b8 100644
--- a/OdbDesignServer/CMakeLists.txt
+++ b/OdbDesignServer/CMakeLists.txt
@@ -58,7 +58,7 @@ foreach(proto_file ${GRPC_PROTO_FILES})
 endforeach()
 
 # Find packages
-find_package(Protobuf CONFIG REQUIRED)
+find_package(Protobuf REQUIRED)
 find_package(gRPC CONFIG REQUIRED)
 
 # Generate protobuf messages for gRPC proto files
diff --git a/OdbDesignServer/Config/GrpcServiceConfig.h b/OdbDesignServer/Config/GrpcServiceConfig.h
index d70b4db..68b75fb 100644
--- a/OdbDesignServer/Config/GrpcServiceConfig.h
+++ b/OdbDesignServer/Config/GrpcServiceConfig.h
@@ -2,7 +2,7 @@
 
 #include <memory>
 #include <string>
-#include <grpc/impl/compression_types.h>
+#include <grpc/compression.h>
 
 namespace OdbDesignServer
 {
diff --git a/OdbDesignServer/Services/OdbDesignServiceImpl.cpp b/OdbDesignServer/Services/OdbDesignServiceImpl.cpp
index a189c0f..4615872 100644
--- a/OdbDesignServer/Services/OdbDesignServiceImpl.cpp
+++ b/OdbDesignServer/Services/OdbDesignServiceImpl.cpp
@@ -4,7 +4,7 @@
 #include <utility>
 #include <odbdesign_export.h>
 #include <featuresfile.pb.h>
-#include <grpcpp/impl/status.h>
+#include <grpcpp/support/status.h>
 #include <string>
 #include <exception>
 #include <grpcpp/support/sync_stream.h>
diff --git a/OdbDesignServer/Services/OdbDesignServiceImpl.h b/OdbDesignServer/Services/OdbDesignServiceImpl.h
index 2662386..bfa936f 100644
--- a/OdbDesignServer/Services/OdbDesignServiceImpl.h
+++ b/OdbDesignServer/Services/OdbDesignServiceImpl.h
@@ -10,7 +10,7 @@
 #include <service.pb.h>
 #include <standardfontsfile.pb.h>
 #include <symbolname.pb.h>
-#include <grpcpp/impl/status.h>
+#include <grpcpp/support/status.h>
 #include <grpcpp/server_context.h>
 #include <grpcpp/support/sync_stream.h>
 
